### PR TITLE
[Modal] Fixed animation issue present in async Modal rendering

### LINKED ISSUE #1897: [Modal] / Missing animation when component is rendered with `open` set to true directly
## Issue summary

`Modal` is not properly animated when rendered with `open` set to true directly. 

### `open && <Modal open ... />` 
Jul-30-2019 14-08-21

### `<Modal open={open}`
Jul-30-2019 14-11-20

For performance optimizations, we often defer the loading of the `Modal` and its content using `createAsyncComponent` from `react-async`. When [following this recipe](https://github.com/Shopify/quilt/blob/master/packages/react-async/documentation/recipes.md#ui-that-is-initially-hidden), you will loose the animation on `Modal`.

## Expected behavior

`Modal` should always be animated when opened, wether the component is directly mounted with `open` set to `true` or `open` is toggled from `false` to `true`.

## Actual behavior

The `Modal` is not properly animated (missing vertical translation).

## Steps to reproduce the problem

See code sandbox link below.

## Reduced test case

https://codesandbox.io/s/ancient-violet-z8te6

## Specifications

- Are you using the React components? (Y/N): Y
- Polaris version number: "@shopify/polaris": "3.20.0",
- Browser: Chrome
- Device: Mac
- Operating System: `10.14.5`

Or run `npx envinfo --system --binaries --browsers --npmPackages react,react-dom,@shopify/polaris` to provide specifications on your environment including version numbers, browser, device, and operating system.

Paste the results here:

```bash

```

### PR DESCRIPTION
**Pull request by mitchmaps**
### Why are these changes introduced?
Fixes #1897 

### What issue is this pull request addressing?
The normal fade up animation present in the modal component currently doesn't appear when setting the component directly to true.

### What is this pull request doing?
This PR addresses this by appending additional `appear` animation classes to the component so its consistent throughout. S/o to @dleroux for suggesting this solution in the issue 🙏

<details>
<summary>Click to view collapsed before and after gifs</summary>

# Before

Jul-30-2019 14-08-21

# After

Jul-30-2019 14-11-20
</details>

If you have any questions reach out to @ mitchell.soares on Slack!

### CODE REVIEW

File: UNRELEASED.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ Use [the changelog guidelines](https://git.io/polaris-changelog-guidelines) to f
 ### Enhancements
 
 ### Bug fixes
+- Fixed animation for Modal when being rendered asynchronously ([#2076](https://github.com/Shopify/polaris-react/pull/2076))
```
Comment (loic-d) [inline UNRELEASED.md]: I would probably just say 'when being mounted with `open:true` directly' or something like that?
Rendering the modal asynchronously using `react-async` is one of the scenario where it's mounted directly with `open:true` but there are others.

File: src/components/Modal/components/Dialog/Dialog.tsx [hunk 2]
```diff
@@ -80,6 +80,8 @@ export function Dialog({
 }
 
 const fadeUpClasses = {
+  appear: classNames(styles.animateFadeUp, styles.entering),
+  appearActive: classNames(styles.animateFadeUp, styles.entered),
```
Comment (loic-d) [inline src/components/Modal/components/Dialog/Dialog.tsx]: ❤️

File: UNRELEASED.md (hunk 1 above)
Comment (mitchmaps) [reply UNRELEASED.md]: Great catch, thanks!

**Comment by mitchmaps** [comment]
Talked to @dleroux about the lack of a matching exit animation brought up by @loic-d above. We agreed to just ship this one that fixes the opening animation error and open a second issue addressing the closing one. 🚢

**Pull request merged**